File: src/Hosting.UI/ClientApp/src/app/shared/enums/date-interval-type.enum.ts
export enum DateIntervalType {
  weekly = 'Weekly',
  monthly = 'Monthly',
  yearly = 'Yearly',
  oneTime = 'OneTime'
}

export const DateIntervalTypeNames = new Map<string, string>([
  [DateIntervalType.weekly, 'Weekly'],
  [DateIntervalType.monthly, 'Monthly'],
  [DateIntervalType.yearly, 'Yearly'],
  [DateIntervalType.oneTime, 'One time']
]);

File: src/Holefeeder.Ui/lib/core/enums/date_interval_type_enum.dart
enum DateIntervalType {
  daily,
  weekly,
  monthly,
  yearly,
  oneTime,
}

extension DateIntervalTypeExtension on DateIntervalType {
  static DateIntervalType fromString(String type) {
    return DateIntervalType.values.firstWhere(
      (e) => e.toString().split('.').last.toLowerCase() == type.toLowerCase(),
      orElse: () => throw ArgumentError('Invalid DateIntervalType: $type'),
    );
  }

  String toStringValue() {
    return toString().split('.').last;
  }
}
